pinot-segment-local/src/test/java/org/apache/pinot/segment/local/recordtransformer/RecordTransformerTest.java

```diff
@@ -178,6 +178,27 @@ public void testSanitationTransformer() {
     }
   }
 
+  @Test
+  public void testLogicalScalarOps() {
+    TableConfig tableConfig = new TableConfigBuilder(TableType.OFFLINE).setTableName("testTable").build();
+
+    // expression true, filtered
+    GenericRow genericRow = getRecord();
+    tableConfig.setIngestionConfig(
+        new IngestionConfig(null, null, new FilterConfig("and(eq(svInt, 123), lte(svInt, 123))"), null, null));
```

Comment:
Does `svInt = 123 AND svInt < 123` work? Users usually don't use `eq` or `lte` explicitly.
Also let's put some filter that can potentially match record to test the behavior

Reply:
Is was LTE. Hence it worked :)
Had added these to be able to test and / or. Made the tests a bit more extensive.
All the filters actually match the record. SKIP_RECORD_KEY is being checked for.